Invalid memory access with libxml2 2.9.9

The stdlib XML specs fail with the following error when linked against libxml2 2.9.9:

```
bin/crystal run --debug spec/std/xml/reader_spec.cr -- -e "#read_outer_xml"
Invalid memory access (signal 11) at address 0x40
[0xaab2f9c880e0] *CallStack::print_backtrace:Int32 +100
[0xaab2f969fb74] __crystal_sigfault_handler +228
[0xfff613bbe964] __setjmp +56
[0xfff613a33170] xmlTextReaderReadOuterXml +20
[0xaab2facf9dbc] *XML::Reader#read_outer_xml:String +24
[0xaab2f9c43b90] ~procProc(Nil) +44
[0xaab2f96c241c] *it<String, String, Int32, Int32, &Proc(Nil)>:(Array(Spec::Result) | Nil) +580
[0xaab2f9c43b54] ~procProc(Nil) +72
[0xaab2f9e1a3a0] *Spec::RootContext::describe<String, String, Int32, &Proc(Nil)>:Spec::Context+ +316
[0xaab2f96c7834] *describe<String, String, Int32, &Proc(Nil)>:Spec::Context+ +8
[0xaab2f9c3d420] ~procProc(Nil) +1024
[0xaab2f9e1a3a0] *Spec::RootContext::describe<String, String, Int32, &Proc(Nil)>:Spec::Context+ +316
[0xaab2f9c44644] *describe<XML::Reader.class, String, Int32, &Proc(Nil)>:Spec::Context+ +48
[0xaab2f968623c] __crystal_main +37812
[0xaab2fa6f3984] *Crystal::main_user_code<Int32, Pointer(Pointer(UInt8))>:Nil +8
[0xaab2fa6f38dc] *Crystal::main<Int32, Pointer(Pointer(UInt8))>:Int32 +40
[0xaab2f968d624] main +8
[0xfff613b93734] ???
make: *** [Makefile:86: spec] Error 11
>>> ERROR: crystal: check failed
```

Tested on Alpine edge with Crystal 0.27.2

With 2.9.8. everything seems to work fine. LibXML2 changelog: https://gitlab.gnome.org/GNOME/libxml2/blob/a7fe7ee45938c53a8dd028dd40baa461191a2fd2/NEWS

This issue was first reported by mps on Gitter and I've been able to reproduce it.

--- a/src/xml/reader.cr
+++ b/src/xml/reader.cr
@@ -129,6 +129,16 @@ struct XML::Reader
 
   # Returns the XML for the node and its content including subtrees.
   def read_outer_xml
+    # On a NONE type libxml2 2.9.9 is giving a segfault:
+    #
+    #   https://gitlab.gnome.org/GNOME/libxml2/issues/43
+    #
+    # so we avoid the issue by returning early here.
+    #
+    # FIXME: if that issue is fixed we should revert this line
+    # to avoid doing an extra C call each time.
+    return "" if node_type.none?
+
     xml = LibXML.xmlTextReaderReadOuterXml(@reader)
     xml ? String.new(xml) : ""
   end

